osxkeychain/osxkeychain.go

```diff
@@ -54,7 +58,16 @@ func (h Osxkeychain) Add(creds *credentials.Credentials) error {
 	errMsg := C.keychain_add(s, label, username, secret)
 	if errMsg != nil {
 		defer C.free(unsafe.Pointer(errMsg))
-		return errors.New(C.GoString(errMsg))
+		switch goMsg := C.GoString(errMsg); goMsg {
+		case errCredentialsAlreadyExist:
+			// If docker login is called in parallel, we may try to
+			// save the same credentials twice. In that case, return
+			// ok, because although we lost the race, the desired
+			// credentials were saved by another process.
+			return nil
```

Comment:
I'm a bit on the fence if this is the right thing to do here. While the related ticket is about the same credentials being stored concurrently, it's also possible that _different_ credentials are being stored for the same registry, in which case an error may be most appropriate.

Perhaps we could add  a `ErrConflict` for this, and a utility to detect that, so that a client could choose to handle it; https://github.com/docker/docker-credential-helpers/blob/db1da9da5d00e82567ef5c29bc819305147e9d38/credentials/error.go#L100-L111


@laurazard any thoughts?;

Reply:
My intuition is that we should handle this a la `EINTR`, i.e. "things happened while your code was blocked on some call and you should handle it".

Otherwise, like @thaJeztah mentioned, we have no guarantees that the credentials saved were the desired ones by this process.

Reply:
@beejeebus what do you think?